remove Bokeh deprecation warnings #1657

vikramnagashoka opened the pull request:
## Description
Replaced the Dash and Cross functions as they are deprecated with Scatter function from the bokeh.models.glyphs module. 

<!--
Thank you so much for your PR! To help us review your contribution, please consider the following points:

- The PR title should summarize the changes, for example "Add new group argument for the pair plot".
  Avoid non-descriptive titles such as "Addresses issue #348".

- The description should provide at least 1-2 sentences describing the pull request in detail and
  link to any relevant issues.

- Please prefix the title of incomplete contributions with [WIP] (to indicate a work in
  progress). WIPs may be useful to (1) indicate you are working on something to avoid
  duplicated work, (2) request broad review of functionality or API, or (3) seek collaborators. -->

## Checklist
<!-- Feel free to remove check-list items aren't relevant to your change -->

- [x] Follows [official](https://github.com/arviz-devs/arviz/blob/main/CONTRIBUTING.md#pull-request-checklist) PR format
- [x] Code style  correct (follows pylint and black guidelines)
- [ ] Changes are listed in [changelog](https://github.com/arviz-devs/arviz/blob/main/CHANGELOG.md#v0xx-unreleased)

<!--
Also, please consider reading the contributing guidelines and code of conduct carefully before submitting the PR. They are available at
- https://github.com/arviz-devs/arviz/blob/main/CONTRIBUTING.md
- https://github.com/arviz-devs/arviz/blob/main/CODE_OF_CONDUCT.md

We understand that PRs can sometimes be overwhelming, especially as the
reviews start coming in. Please let us know if the reviews are unclear or
the recommended next step seems overly demanding, if you would like help in
addressing a reviewer's comments, or if you have been waiting too long to hear
back on your PR.
-->

OriolAbril commented on arviz/plots/backends/bokeh/elpdplot.py:
```diff
@@ -163,9 +163,7 @@ def _plot_atomic_elpd(
-    glyph = marker_func(
-        x="xdata", y="ydata", size="sizes", line_color=plot_kwargs.get("color", "black")
-    )
+    glyph = Scatter(x="xdata", y="ydata", size="sizes", line_color=plot_kwargs.get("color", "black"), marker='cross')
```
note that while the other instances call `Dash` explicitly, here the `marker_func` is used instead, which allows users to choose the marker to be used. I believe the suggestion will fix this change in funcitonality, marker is already defined correctly, but you should definitely try it out locally

```suggestion
    glyph = Scatter(x="xdata", y="ydata", size="sizes", line_color=plot_kwargs.get("color", "black"), marker=marker)
```

OriolAbril reviewed (changes requested):
Thanks for the PR

vikramnagashoka reviewed (commented):
Updated to use marker variable instead of constant value of 'cross'

OriolAbril commented:
You have to make sure that all tests pass. Do you need help interpreting the ci errors?

vikramnagashoka commented:
Yes please. I see that the error is related to pylint. The output from this task says score is 10/10. Also when I navigated to the azure pipeline it said there were other jobs which failed with same error.

OriolAbril commented:
You have to take a look at the messages at https://dev.azure.com/ArviZ/ArviZ/_build/results?buildId=4342&view=logs&j=af9a1d3a-ff8f-5483-6823-86518c773a9b&t=91d5999d-c3e7-502d-15b0-1abfa51eab6f&l=14, I never really understood the score out of 10 that gets assigned. What's important is to fix all the messages, even if it says 10, if there are error messages, there are things wrong and CI will therefore fail. The errors are related to lines too long (running black should fix this) and to extra code that is no longer needed with the `Scatter` instead of `marker_fun` change

OriolAbril commented:
It looks like you have modified many unrelated files and now black is complaining. Could it be that you ran black but you have an old version of black installed?

vikramnagashoka commented:
@OriolAbril not sure what you mean by unrelated files. All the files that I changed had either the Dash or Cross functions which I replaced with Scatter. Can you please let me know what I'm missing here ? 

The black version I'm using is as below. 
_**black, version 21.4b2**_

Can you please help me resolve this ?

OriolAbril commented:
I must have been looking at a wrong commit or misread the filenames, sorry. CI is currently failing at the black step: https://dev.azure.com/ArviZ/ArviZ/_build/results?buildId=4344&view=logs&j=af9a1d3a-ff8f-5483-6823-86518c773a9b&t=6a1cfc46-f4a4-5e3c-e2fb-18ea29f733bf&l=14. It looks like (https://dev.azure.com/ArviZ/ArviZ/_build/results?buildId=4344&view=logs&j=af9a1d3a-ff8f-5483-6823-86518c773a9b&t=0aaa11b3-1b3c-5ed3-0c54-8fd8e63dd2e8&l=25) you have the same version installed as the one running on CI, so you should run black and commit the changes to fix this. The command to run black on ArviZ is on the PR checklist: https://github.com/arviz-devs/arviz/blob/main/CONTRIBUTING.md#pull-request-checklist

vikramnagashoka commented:
@OriolAbril thanks for the details. I have pushed changes after running black and now all checks pass. Can you please help me merge the branch ?

OriolAbril reviewed (approved):
Thanks for updating and fixing all the tests, you should add the fixes to the changelog in the "maintenence and fixes" section and we'll merge the PR :smile: 

If you are worried about the PR not being merged eventually I can already guarantee it will be merged

OriolAbril commented:
Thanks @vikramnagashoka
